content/docs/installation/best-practice.md

```diff
@@ -22,6 +22,174 @@ are designed for backwards compatibility rather than for best practice or maximu
+
+There is little justification for using more than 2 replicas of these components.
+Further replicas *may* add a degree of resilience
+if you have the luxury of sufficient Nodes
+with sufficient CPU and memory to accommodate the standby replicas.
+
+### webhook
+
+By default the cert-manager webhook Deployment has 1 replica, but in production you should use 3 or more.
+If the cert-manager webhook is unavailable, all API operations on cert-manager custom resources will fail,
+and this will disrupt any software that creates, updates or deletes cert-manager custom resources.
+So it is *especially* important to keep at least one replica of the cert-manager webhook running at all times.
+
+> ℹ️ By contrast, if there is only a single replica of the cert-manager controller, there is less risk of disruption.
+> For example, if the Node hosting the single cert-manager controller manager Pod is drained,
+> there will be a delay while a new Pod is started on another Node,
+> and any cert-manager resources that are created or changed during that time will not be reconciled until the new Pod starts up.
+> But the controller manager works asynchronously anyway, so any applications which depend on the cert-manager custom resources
+> will be designed to tolerate this situation.
+> That being said, the best practice is to run 2 or more replicas of each controller if the cluster has sufficient resources.
+
+### Topology Spread Constraints
+
+Ensure that a disruption of a node or data center does not degrade the operation of cert-manager,
+by configuring [Topology Spread Constraints](https://kubernetes.io/docs/concepts/scheduling-eviction/topology-spread-constraints/)
+for each of the components.
+For example, the following Helm chart values add topology spread constraints for all three long-running components,
+to request (but not require) Kubernetes to avoid scheduling Pods of the same Deployment to the same zone or node.
+
+```yaml
+topologySpreadConstraints:
+- maxSkew: 1
+  topologyKey: topology.kubernetes.io/zone
+  whenUnsatisfiable: ScheduleAnyway
+  labelSelector:
+    matchLabels:
+      app.kubernetes.io/instance: cert-manager
+      app.kubernetes.io/component: controller
+- maxSkew: 1
+  topologyKey: kubernetes.io/hostname
+  whenUnsatisfiable: ScheduleAnyway
+  labelSelector:
+    matchLabels:
+      app.kubernetes.io/instance: cert-manager
+      app.kubernetes.io/component: controller
+
+webhook:
+  replicaCount: 3
+  topologySpreadConstraints:
+  - maxSkew: 1
+    topologyKey: topology.kubernetes.io/zone
+    whenUnsatisfiable: ScheduleAnyway
+    labelSelector:
+      matchLabels:
+        app.kubernetes.io/instance: cert-manager
+        app.kubernetes.io/component: webhook
+  - maxSkew: 1
+    topologyKey: kubernetes.io/hostname
+    whenUnsatisfiable: ScheduleAnyway
+    labelSelector:
+      matchLabels:
+        app.kubernetes.io/instance: cert-manager
+        app.kubernetes.io/component: webhook
```

Comment:
I'm easy here, not having it is less YAML.
Having it means it's explicit.

Reply:
I think if we can verify that this works by default, we should not tell people to configure this (because we want to keep our docs as simple as possible). Instead, we can just link to the documentation you shared and say that this works correctly by default.

Reply:
I agree that according to the docs the desired anti-affinity scheduling should happen by default.
I've updated this paragraph.